indexing-service/src/main/java/io/druid/indexing/common/task/Task.java

```diff
@@ -128,6 +135,20 @@
   public boolean isReady(TaskActionClient taskActionClient) throws Exception;
 
   /**
+   * Returns whether or not this task can restore its progress from its on-disk working directory. Restorable tasks
+   * may be started with a non-empty working directory. Tasks that exit uncleanly may still have a chance to attempt
+   * restores, meaning that restorable tasks should be able to deal with potentially partially written on-disk state.
+   */
+  public boolean canRestore();
```

Comment:
By 'persistent resources' do you mean a persistent working directory, so if the machine running the task fails, it can be restored from a different machine with the original persistent working directory?

If so… that would be awesome.